### PR TITLE
detect/port: fix grouping of ports w gaps

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/6881
`master`
```
real	0m29.752s
user	0m29.283s
sys	0m0.441s
```
vs this PR
```
real	0m29.670s
user	0m29.213s
sys	0m0.430s
```

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1720

### CODE REVIEW

File: src/detect-engine-build.c [hunk 1]
```diff
@@ -1425,7 +1425,12 @@ static inline int CreatePortList(DetectEngineCtx *de_ctx, const uint8_t *unique_
                 port = port2 + 1;
             } else if (p1 && p1->single) {
                 SCPortIntervalFindOverlappingRanges(de_ctx, port, port, &it->tree, list);
```
Comment (catenacyber) [inline src/detect-engine-build.c:1427]: Do we still need this `SCPortIntervalFindOverlappingRanges` ?

**Comment by catenacyber** [comment]
Why is this a draft ?

File: src/detect-engine-build.c (hunk 1 above)
Comment (inashivb) [reply src/detect-engine-build.c:1427]: yes. This looks for [port, port]. Next one looks for [port, port2 -1]

**Comment by inashivb** [comment]
> Why is this a draft ?

Waiting to see if it passes QA, will set it "Ready for review" then

**Pull request closed without merging**